src/content/service.js

```diff
@@ -122,161 +164,142 @@ module.exports = ({ db, botfile, projectLocation, logger }) => {
-    const knex = await db.get()
+  const loadCategory = file => {
+    const filePath = path.resolve(formDir, './' + file)
+    // eslint-disable-next-line no-eval
+    const category = eval('require')(filePath) // Dynamic loading require eval for Webpack
+    const requiredFields = ['id', 'title', 'jsonSchema']
+
+    requiredFields.forEach(field => {
+      if (_.isNil(category[field])) {
+        throw new Error(field + ' is required but missing in Content Form file: ' + file)
+      }
+    })
 
-    let query = knex('content_items').select('*')
+    category.id = category.id.toLowerCase()
 
-    if (ids) {
-      query = query.whereIn('id', ids)
+    if (categoryById[category.id]) {
+      throw new Error('There is already a form with id=' + category.id)
     }
 
-    const items = query.then()
+    categoryById[category.id] = category
+    categories.push(category)
 
-    return items.map(item => transformCategoryItem(item))
+    return category
   }
 
-  const importContent = async documents => {
-    const knex = await db.get()
-
-    return Promise.mapSeries(documents, doc => {
-      if (!doc.id || !doc.formData || !doc.categoryId) {
-        throw new Error('Invalid data')
+  const readDataFromFile = (filePath, fileName) => {
+    if (!fs.existsSync(filePath)) {
+      logger.warn(`Form content file ${filePath} not found`)
+      return []
+    }
+    try {
+      const json = fs.readFileSync(filePath, 'utf-8')
+      const data = JSON.parse(json)
+      if (!Array.isArray(data)) {
+        throw new Error(`${fileName} expected to contain array, contents ignored`)
       }
+      logger.info(`Read ${data.length} item(s) from ${fileName}`)
+      return data
+    } catch (err) {
+      logger.warn(`Error reading data from ${fileName}`, err)
+      return []
+    }
+  }
 
-      const row = {
-        data: JSON.stringify(doc.data),
-        formData: JSON.stringify(doc.formData),
-        metadata: '|' + doc.metadata.join('|') + '|',
-        previewText: doc.previewText,
-        created_by: 'admin',
-        created_on: helpers(knex).date.now(),
-        id: doc.id,
-        categoryId: doc.categoryId
-      }
+  const loadData = async (category, fileName) => {
+    const filePath = path.resolve(formDataDir, './' + fileName)
+    fileById[category.id] = filePath
 
-      return knex('content_items')
-        .insert(row)
-        .then()
-        .catch(err => {
-          return knex('content_items')
-            .where({ id: doc.id })
-            .update(row)
-            .then()
-        })
+    logger.debug(`Loading data for ${category.id} from ${fileName}`)
+    let data = []
+    try {
+      data = readDataFromFile(filePath, fileName)
+    } catch (e) {}
+
+    data.forEach(item => {
+      item.categoryId = category.id
+      item.id = item.id || getNewItemId(category)
+    })
+
+    return await Promise.map(data, async item => await knex('content_items').insert(transformItemApiToDb(item)))
```

Comment:
This reminds me we should add a TODO for using a transaction here

Reply:
For in-memory DB this should be less important, but still yeah, good to have